core/modules/system/system.admin.inc

```diff
@@ -2792,10 +2792,20 @@ function system_transliteration_retroactive_submit($form, &$form_state) {
  */
 function system_transliteration_file_query() {
   $operator = 'NOT REGEXP';
+  $regex = '/[a-z0-9_.-]+$';
+
   if (config_get('system.core', 'file_transliterate_lowercase')) {
-    $operator .= ' BINARY';
+    // As of version 8.0.22 MySQL throws an error on binary comparison, but
+    // recent versions support setting a flag for case sensitive comparison in
+    // the regex.
+    $connection = Database::getConnection();
+    if ($connection->databaseType() == 'mysql' && version_compare($connection->version(), '8.0.22', '>=')) {
+      $regex = '/(?-i)[a-z0-9_.-]+$';
```

Comment:
Should we do something like this here:

```php
  $regex = '[a-z0-9_.-]+';
  ...
  if (...) {
      $regex = '(?-i)' . $regex;
  }
  else {
    ...
  }
  $regex = '/' . $regex . '$';
```

...so that if the "basic" regex ever needs to be changed, we only update it in a single place.

Reply:
I feel like this makes the code less readable - could be wrong, though. Personally, I'd prefer to see the full regex pattern in use, not concatenated.

Reply:
I agree, better all together.